File: src/cxx/Vulkan/VulkanImmediateShaderData/PushConstants/VulkanPushConstant.cpp
#include "VulkanPushConstant.h"

VulkanPushConstant::VulkanPushConstant(size_t size, VkShaderStageFlags shaderStages) : size(size), shaderStages(shaderStages)
{
}

size_t VulkanPushConstant::getSize()
{
    return size;
}

VkShaderStageFlags VulkanPushConstant::getShaderStages()
{
    return shaderStages;
}

void *VulkanPushConstant::getData()
{
    return data;
}

void VulkanPushConstant::setData(void *data)
{
    VulkanPushConstant::data = data;
}

File: src/main/java/com/kgaft/VulkanContext/Vulkan/VulkanImmediateShaderData/VulkanPushConstants/VulkanPushConstant.java
package com.kgaft.VulkanContext.Vulkan.VulkanImmediateShaderData.VulkanPushConstants;

import java.nio.ByteBuffer;

public class VulkanPushConstant {
   private long size;
    private int shaderStages;
    private ByteBuffer data;
    
    public VulkanPushConstant(long size, int shaderStages){
        this.size = size;
        this.shaderStages = shaderStages;
        data = ByteBuffer.allocateDirect((int)size);
    }

    public long getSize() {
        return size;
    }

    public int getShaderStages() {
        return shaderStages;
    }

    public ByteBuffer getData() {
        return data;
    }
    
    
}
